WF-CDPInstaller/plugin_dir/plugin_installer.py

```diff
@@ -0,0 +1,186 @@
+import common.install_utils as utils
+import common.config as config
+import plugin_dir.plugin_exception as ex
+
+
+class PluginInstaller(object):
+    """
+    The interface for installers
+
+    The following methods need to be implemented
+      title():
+          - the ascii art for the installer
+      overview():
+          - general description about what the installer does
+      check_depedency()
+          - check plugin and installer's dependency
+      write_plugin():
+          - write the actual plugin file
```

Comment:
as @kentwang929 and I discussed, this will be fine for this version, but in the next edition, `write_plugin()` will be broken up into 2 phases, `gather_data()`, which will gather and validate the data interactively from the user, and `output_config()`, which actually writes out the file.  They will pass a dict of the collected values between them.